File: benchmark/sum_squares.c
#include "bench_util.h"

int main(void) {
  const int64_t n = bench_env_i64("BENCH_SUM_SQUARES_N", 80000000);
  const int64_t mod = 1000000007;
  int64_t sum = 0;
  int64_t t0 = bench_now_ms();
  for (int64_t i = 1; i <= n; ++i) {
    int64_t sq = i * i;
    sum = (sum + sq) % mod;
    if (sum < 0)
      sum += mod;
  }
  int64_t ms = bench_now_ms() - t0;
  bench_print_line("C", "sum_squares", ms, (uint64_t)sum);
  return 0;
}

File: benchmark/sum_squares.py
#!/usr/bin/env python3
import os
import time

N = int(os.environ.get("BENCH_SUM_SQUARES_N", "80000000"))
MOD = 1_000_000_007

def sum_squares() -> int:
    s = 0
    for i in range(1, N + 1):
        s = (s + i * i) % MOD
    return s

def main() -> None:
    t0 = time.perf_counter()
    result = sum_squares()
    ms = int((time.perf_counter() - t0) * 1000)
    print(f"{'Python':<8} {'sum_squares':<14} {ms:8d} ms  result={result}")

if __name__ == "__main__":
    main()
